datadotworld/client/content_negotiating_api_client.py

```diff
@@ -18,3 +19,15 @@ def select_header_accept(self, accepts):
             return self.default_mimetype
         else:
             return ', '.join(accepts)
+
+    def select_header_content_type(self, content_types):
+
+        if not content_types:
+            return 'application/json'
+
+        content_types = [x.lower() for x in content_types]
+
+        if self.default_mimetype in content_types:
+            return self.default_mimetype
```

Comment:
A different parameter should driver `Accept` and `Content-Type` headers. Have you tried making an `append_records()` call using `application/json-l`?

Reply:
Updated. Yes, I tried this out and it works. The content-type will always be `application/json` except a user specifies `application/json-l`.